### PR TITLE
🚨🚨 Source Amazon Seller Partner: remove brand analytics reports from cloud

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve https://github.com/airbytehq/alpha-beta-issues/issues/1741 and https://github.com/airbytehq/alpha-beta-issues/issues/1493

## How

Remove brand analytics reports from cloud, permanently remove [deprecated](https://developer-docs.amazon.com/sp-api/changelog/january-2023-release-announcement) FBA reports

## Recommended reading order
1. `source.py`
2. `streams.py`

## 🚨 User Impact 🚨

Removed streams: 

- `GET_BRAND_ANALYTICS_MARKET_BASKET_REPORT`
- `GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT`
- `GET_BRAND_ANALYTICS_REPEAT_PURCHASE_REPORT`
- `GET_BRAND_ANALYTICS_ALTERNATE_PURCHASE_REPORT`
- `GET_BRAND_ANALYTICS_ITEM_COMPARISON_REPORT`
- `GET_SALES_AND_TRAFFIC_REPORT`
- `GET_VENDOR_SALES_REPORT`
- `GET_VENDOR_INVENTORY_REPORT`
- `GET_FBA_FULFILLMENT_INVENTORY_ADJUSTMENTS_DATA`
- `GET_FBA_FULFILLMENT_CURRENT_INVENTORY_DATA`
- `GET_FBA_FULFILLMENT_INVENTORY_RECEIPTS_DATA`
- `GET_FBA_FULFILLMENT_INVENTORY_SUMMARY_DATA`
- `GET_FBA_FULFILLMENT_MONTHLY_INVENTORY_DATA`

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: docs/integrations/sources/amazon-seller-partner.md [hunk 1]
```diff
@@ -82,12 +82,7 @@ This source is capable of syncing the following tables and their data:
 - [Financial Event Groups](https://developer-docs.amazon.com/sp-api/docs/finances-api-reference#get-financesv0financialeventgroups)
 - [Financial Events](https://developer-docs.amazon.com/sp-api/docs/finances-api-reference#get-financesv0financialevents)
 - [FBA Fee Preview Report](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference)
-- [FBA Daily Inventory History Report](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference)
 - [FBA Promotions Report](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference)
-- [FBA Inventory Adjustments Report](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference)
-- [FBA Received Inventory Report](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference)
-- [FBA Inventory Event Detail Report](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference)
-- [FBA Monthly Inventory History Report](https://developer-docs.amazon.com/sp-api/docs/reports-api-v2021-06-30-reference)
```
Comment (lazebnyi) [inline docs/integrations/sources/amazon-seller-partner.md:90]: Need mention that those stream presented only in oss

**Review by lazebnyi — APPROVED** [review]
LGTM 
Small comment to doc.

File: docs/integrations/sources/amazon-seller-partner.md (hunk 1 above)
Comment (askarpets) [reply docs/integrations/sources/amazon-seller-partner.md:90]: Notes have been added

File: airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml [hunk 2]
```diff
@@ -20,6 +20,11 @@ data:
     oss:
       enabled: true
   releaseStage: alpha
+  releases:
+    breakingChanges:
+      2.0.0:
+        message: Brand Analytics Reports will be removed from Cloud verison. Deprecated FBA reports will be removed permanently.
+        upgradeDeadline: "2023-11-24"
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml:27]: let's make the date 11-29 we have an american holiday next week

File: airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml [hunk 3]
```diff
@@ -20,6 +20,11 @@ data:
     oss:
       enabled: true
   releaseStage: alpha
+  releases:
+    breakingChanges:
+      2.0.0:
+        message: Brand Analytics Reports will be removed from Cloud verison. Deprecated FBA reports will be removed permanently.
```
Comment (katmarkham) [inline airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml:26]: can we talk about why/what the benefit is?

File: docs/integrations/sources/amazon-seller-partner-migrations.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Amazon Seller Partner Migration Guide
+
+## Upgrading to 2.0.0
+
+This change removes Brand Analytics from Airbyte Cloud version of connector and permanently removes deprecated FBA reports.
+Customers who have those streams must refresh their schema OR disable next streams:
```
Comment (katmarkham) [inline docs/integrations/sources/amazon-seller-partner-migrations.md:6]: Customers who have those streams must refresh their schema OR disable the following streams:

File: docs/integrations/sources/amazon-seller-partner-migrations.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
+# Amazon Seller Partner Migration Guide
+
+## Upgrading to 2.0.0
+
+This change removes Brand Analytics from Airbyte Cloud version of connector and permanently removes deprecated FBA reports.
```
Comment (katmarkham) [inline docs/integrations/sources/amazon-seller-partner-migrations.md:5]: This change removes Brand Analytics and permanently removes deprecated FBA reports (from Airbyte Cloud).

**Review by katmarkham — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/amazon-seller-partner-migrations.md (hunk 5 above)
Comment (artem1205) [reply docs/integrations/sources/amazon-seller-partner-migrations.md:5]: To be more precise:
Brand Analytics removed from CLOUD version only (will still be available for OSS users).
FBA reports removed for both CLOUD and OSS.

Which comment should be left?

File: airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml (hunk 3 above)
Comment (katmarkham) [reply airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml:26]: Deprecated FBA reports will be removed permanently from Cloud and Brand Analytics Reports will be removed temporarily.  Updates on Brand Analytics Reports can be tracked here: https://github.com/airbytehq/airbyte/issues/32353

File: airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml (hunk 2 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-amazon-seller-partner/metadata.yaml]: done

File: docs/integrations/sources/amazon-seller-partner-migrations.md (hunk 4 above)
Comment (artem1205) [reply docs/integrations/sources/amazon-seller-partner-migrations.md]: done

File: docs/integrations/sources/amazon-seller-partner-migrations.md (hunk 5 above)
Comment (artem1205) [reply docs/integrations/sources/amazon-seller-partner-migrations.md]: done

**Review by katmarkham — APPROVED** [review]

**Comment by kodeine** [comment]
any idea's when brand analytics reports will be added again?

**Pull request merged**